cmd/nerdctl/image/image_list_test.go

```diff
@@ -117,6 +118,13 @@ func TestImages(t *testing.T) {
 		},
 	}
 
+	if runtime.GOOS == "windows" {
+		testCase.Require = test.Require(
+			testCase.Require,
+			nerdtest.IsFlaky("https://github.com/containerd/nerdctl/issues/3524"),
+		)
```

Comment:
can you explain this a bit?

Reply:
Sure.

On windows, this may fail (randomly) - detailed in the linked ticket.

`nerdtest.IsFlaky("issue link")` is part of the new test framework (https://github.com/containerd/nerdctl/blob/main/docs/testing/tools.md#requirements)

It marks a test as flaky, which makes it run only if `-test.only-flaky` is provided to the integration test.
Very soon, with PR #3535 we will have separate steps to run non-flaky (no retries, hard failure) and flaky tests (retries, hard fail only if all retries fail).

Now, this test also has a requirement of `not(docker)`: https://github.com/containerd/nerdctl/blob/ce8e077d9548277ea05e543565cb7d33b7e57e65/cmd/nerdctl/image/image_list_test.go#L40
and I do not want that to get overridden here (otherwise, if ran on windows against target docker, the test would not be skipped).

`test.Require(reqs...)` is a helper that just bundles together a set of requirements (the previous one `testCase.Require`, and the conditional one `IsFlaky`).

Lmk if that does not clarify or if your question was about something I missed?

Thanks @djdongjin